src/Test/Utilities/Portable/Assert/ConditionalFactAttribute.cs

```diff
@@ -166,6 +166,7 @@ public static class ExecutionConditionUtil
         public static bool IsCoreClr => !IsDesktop;
         public static bool IsCoreClrUnix => IsCoreClr && IsUnix;
         public static bool IsMonoOrCoreClr => IsMono || IsCoreClr;
+        public static bool RuntimeSupportsCovariantReturnsOfClasses => Type.GetType("System.Runtime.CompilerServices.RuntimeFeature")?.GetField("CovariantReturnsOfClasses") != null;
```

Comment:
>RuntimeSupportsCovariantReturnsOfClasses [](start = 27, length = 40)

This feels wrong. It shouldn't matter in which runtime a test is executed. A test should be able any Target framework for a compilation and only the actual references should determine whether covariant returns are supported within the compilation.

Reply:
These are testing runtime conditions. Running covariant tests does depend on having a runtime that supports them.

---
In reply to: [461778705](https://github.com/dotnet/roslyn/pull/46367#discussion_r461778705) [](ancestors = 461778705)

Reply:
> Running covariant tests does depend on having a runtime that supports them.

Executing test compiled code that utilizes covariant return feature - yes. Everything else, shape of symbols, result of emit, diagnostics, etc. doesn't depend

---
In reply to: [461836199](https://github.com/dotnet/roslyn/pull/46367#discussion_r461836199) [](ancestors = 461836199,461778705)

Reply:
Right, which is why tests that are designed to test other things are not conditional.

---
In reply to: [461946462](https://github.com/dotnet/roslyn/pull/46367#discussion_r461946462) [](ancestors = 461946462,461836199,461778705)